File: backend/src/main/java/cliquee/backend/model/Block.java
package cliquee.backend.model;

import cliquee.backend.model.embedded.BlockId;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.Data;

@Data
@Entity
@Table(name = "blocks")
public class Block {

  @EmbeddedId
  private BlockId id;

  @ManyToOne(fetch = FetchType.LAZY)
  @MapsId("blocker_id")
  @JoinColumn(name = "blocker_id")
  private User blocker;

  @ManyToOne(fetch = FetchType.LAZY)
  @MapsId("blocked_id")
  @JoinColumn(name = "blocked_id")
  private User blocked;

  private LocalDateTime created_at = LocalDateTime.now();
}

File: flaskBackend/app/models/block.py
from datetime import datetime
from ..extensions import db

class Block(db.Model):
    __tablename__ = 'blocks'
    blocker_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False, primary_key=True)
    blocked_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False, primary_key=True)
    created_at = db.Column(db.DateTime, default=datetime.now())

    def __repr__(self):
        return f'<Block {self.blocker_id} blocked {self.blocked_id}>'